p2p/conn/secret_connection.go

```diff
@@ -87,7 +92,15 @@ func MakeSecretConnection(conn io.ReadWriteCloser, locPrivKey crypto.PrivKey) (*
 	}
 
 	// Sort by lexical order.
-	loEphPub, _ := sort32(locEphPub, remEphPub)
+	loEphPub, hiEphPub := sort32(locEphPub, remEphPub)
+
+	io.ReadFull(hkdfInit, hdkf_state[:])
```

Comment:
`io.ReadFull` can error. Hence, this should be
```suggestion
	if err := io.ReadFull(hkdfInit, hdkf_state[:]); err != nil {
	  return nil, err
	}
```